tfhe/benches/integer/zk_pke.rs

```diff
@@ -176,82 +215,142 @@ fn pke_zk_verify(c: &mut Criterion, results_file: &Path) {
+                        );
+
+                        let test_name = format!(
+                            "zk::proven_list_size::{param_name}_{bits}_bits_packed_{zk_load}"
+                        );
+
+                        write_result(
+                            &mut file,
+                            &test_name,
+                            proven_ciphertext_list_serialized.len(),
+                        );
+                        write_to_json::<u64, _>(
+                            &test_name,
+                            shortint_params,
+                            param_name,
+                            "pke_zk_proof",
+                            &OperatorType::Atomic,
+                            0,
+                            vec![],
+                        );
+
+                        let proof_size = ct1.proof_size();
+                        println!("proof size: {}", ct1.proof_size());
+
+                        let test_name =
+                            format!("zk::proof_sizes::{param_name}_{bits}_bits_packed_{zk_load}");
+
+                        write_result(&mut file, &test_name, proof_size);
+                        write_to_json::<u64, _>(
+                            &test_name,
+                            shortint_params,
+                            param_name,
+                            "pke_zk_proof",
+                            &OperatorType::Atomic,
+                            0,
+                            vec![],
+                        );
+
+                        bench_group.bench_function(&bench_id_verify, |b| {
+                            b.iter(|| {
+                                let _ret = ct1.verify(public_params, &pk, &metadata);
+                            });
+                        });
+
+                        bench_group.bench_function(&bench_id_verify_and_expand, |b| {
+                            b.iter(|| {
+                                let _ret = ct1
+                                    .verify_and_expand(
+                                        public_params,
+                                        &pk,
+                                        &metadata,
+                                        IntegerCompactCiphertextListExpansionMode::CastAndUnpackIfNecessary(
+                                            casting_key.as_view(),
+                                        ),
+                                    )
+                                    .unwrap();
+                            });
+                        });
+                    }
+                    BenchmarkType::Throughput => {
+                        // In throughput mode object sizes are not recorded.
+                        let elements = throughput_num_threads();
+                        bench_group.throughput(Throughput::Elements(elements)); // FIXME CPU is not fully loaded in the current state (around 75% on my
+                        // laptop)
```

Comment:
did you ever manage to get this to saturate the CPU ?

Reply:
Yes, after investigation, it's just the first bits size iteration (`64usize`) which never saturate the CPU.
Following elements in the iterator saturate the CPU just fine.

Reply:
what do you mean by first bit size iteration ?